src/rdkafka_admin.c

```diff
@@ -0,0 +1,2696 @@
+ *     rko on the rdkafka main ops queue again we go to back to step 8
+ *     to check if broker is now available.
+ *
+ * 11. [rdkafka main thread] Back in the worker callback we now have an
+ *     rkb broker pointer (with reference count increased) for the controller
+ *     with the connection up (it might go down while we're referencing it,
+ *     but that does not stop us from enqueuing a protocol request).
+ *
+ * 12. [rdkafka main thread] A DeleteTopics protocol request buffer is
+ *     constructed using the input parameters saved on the rko and the
+ *     buffer is enqueued on the broker's transmit queue.
+ *     The buffer is set up to provide the reply buffer on the rdkafka main
+ *     ops queue (the same queue we are operating from) with a handler
+ *     callback of rd_kafka_admin_handle_response().
+ *     The state is updated to the RD_KAFKA_ADMIN_STATE_WAIT_RESPONSE.
+ *
+ * 13. [broker thread] If the request times out, a response with error code
+ *     (ERR__TIMED_OUT) is enqueued. Go to 16.
+ *
+ * 14. [broker thread] If a response is received, the response buffer
+ *     is enqueued. Go to 16.
+ *
+ * 15. [rdkafka main thread] The buffer callback (..handle_response())
+ *     is called, which attempts to extra the original rko from the eonce,
+ *     but if the eonce has already been triggered by some other source
+ *     (the timeout timer) the buffer callback simply returns and does nothing
+ *     since the admin request is over and a result (probably a timeout)
+ *     has been enqueued for the application.
+ *     If the rko was still intact we temporarily set the reply buffer
+ *     in the rko struct and call the worker callback. Go to 17.
+ *
+ * 16. [rdkafka main thread] The worker callback is called in state
+ *     RD_KAFKA_ADMIN_STATE_WAIT_RESPONSE without a response but with an error.
+ *     An error result op is created and enqueued on the application's
+ *     provided response rkqu queue.
+ *     
+ * 17. [rdkafka main thread] The worker callback is called in state
+ *     RD_KAFKA_ADMIN_STATE_WAIT_RESPONSE with a response buffer with no
+ *     error set.
+ *     The worker calls the response `parse()` callback to parse the response
+ *     buffer and populates a result op (rko_result) with the response
+ *     information (such as per-topic error codes, etc).
+ *     The result op is returned to the worker.
+ *
+ * 18. [rdkafka main thread] The worker enqueues the result up (rko_result)
+ *     on the application's provided response rkqu queue.
+ *
+ * 19. [app thread] The application calls rd_kafka_queue_poll() to
+ *     receive the result of the operation. The result may have been
+ *     enqueued in step 18 thanks to succesful completion, or in any
+ *     of the earlier stages when an error was encountered.
+ *
+ * 20. [app thread] The application uses rd_kafka_event_DeleteTopics_result()
+ *     to retrieve the request-specific result type.
+ *
+ * 21. Done, so easy.
```

Comment:
you might want to consider rewording # 21

Reply:
:(